src/main/java/ch/njol/skript/expressions/base/EventValueExpression.java

```diff
@@ -77,20 +79,27 @@ public EventValueExpression(final Class<? extends T> c, final @Nullable Changer<
 		assert c != null;
 		this.c = c;
 		this.changer = changer;
+		single = !c.isArray();
+		componentType = single ? c : c.getComponentType();
 	}
 	
 	@Override
 	@Nullable
+	@SuppressWarnings({"unchecked", "SuspiciousSystemArraycopy"})
 	protected T[] get(final Event e) {
 		final T o = getValue(e);
 		if (o == null)
-			return null;
-		@SuppressWarnings("unchecked")
-		final T[] one = (T[]) Array.newInstance(c, 1);
-		one[0] = o;
-		return one;
+			return (T[]) new Object[0];
```

Comment:
```suggestion
			return (T[]) Array.newInstance(c, 0);
```
I would keep the usage of Array.newInstance for clarity.